<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="central_widget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>3</x>
      <y>3</y>
      <width>791</width>
      <height>551</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="main_layout" stretch="1,3">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>50</number>
     </property>
     <item>
      <widget class="QFrame" name="sidebar_frame">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>92</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>161</width>
          <height>331</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="sidebar_layout" stretch="1,3,1,0,0">
         <item alignment="Qt::AlignHCenter">
          <widget class="QLabel" name="topology_label">
           <property name="text">
            <string>Topology</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTabWidget" name="create_helper">
           <property name="currentIndex">
            <number>1</number>
           </property>
           <widget class="QWidget" name="tab">
            <property name="toolTip">
             <string/>
            </property>
            <property name="toolTipDuration">
             <number>-1</number>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <attribute name="title">
             <string>Host</string>
            </attribute>
            <widget class="QWidget" name="layoutWidget">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>10</y>
               <width>131</width>
               <height>81</height>
              </rect>
             </property>
             <layout class="QVBoxLayout" name="button_layout">
              <property name="spacing">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="add_host_button">
                <property name="text">
                 <string>Add Host</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="remove_host_button">
                <property name="text">
                 <string>Remove Host</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
           <widget class="QWidget" name="tab_2">
            <attribute name="title">
             <string>Switch</string>
            </attribute>
            <widget class="QWidget" name="layoutWidget_2">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>10</y>
               <width>131</width>
               <height>81</height>
              </rect>
             </property>
             <layout class="QVBoxLayout" name="button_layout_2">
              <property name="spacing">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="add_switch_button">
                <property name="text">
                 <string>Add Switch</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="remove_switch_button">
                <property name="text">
                 <string>Remove Switch</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="flows_label">
           <property name="text">
            <string>Flows</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="add_flow_btn">
           <property name="text">
            <string>Add Flow</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="remove_flow_btn">
           <property name="text">
            <string>Remove Flow</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QPushButton" name="refresh_button">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>500</y>
          <width>159</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>Refresh</string>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="CanvasWidget" name="canvas_widget" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionAdd_Host">
   <property name="text">
    <string>Add Host</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CanvasWidget</class>
   <extends>QWidget</extends>
   <header>canvas_widget</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
